subjects/ai/numpy/audit/README.md

```diff
@@ -140,180 +193,191 @@ integers[mask] = 0
-
-Here is an example of a possible solution:
-
-```python
-        x[1:8,1:8] = 0
-        x[2:7,2:7] = 1
-        x[3:6,3:6] = 0
-        x[4,4] = 1
 ```
 
-###### For question 3, is the output the following?
+##### Check the solution for cheating like:
 
-```console
-   array([[ 1,  2,  3],
-          [ 2,  4,  6],
-          [ 3,  6,  9],
-          [ 4,  8, 12],
-          [ 5, 10, 15]], dtype=int8)
-```
-
-##### The solution of question 3 is not accepted if the values of the array have been changed one by one manually. The usage of the for loop is not allowed neither.
+- The values of the array have been changed one by one manually.
+- The usage of the for loop, which is not allowed.
+- Printing the full output given in the readme.
 
-Here is an example of a possible solution:
-
-```python
-    np.reshape(arr_1, (5, 1)) * arr_2
-```
+###### Can you confirm that there was no cheating in the solution?
 
 ---
 
 ---
 
 #### Exercise 7: NaN
 
-##### The exercise is validated if all questions of the exercise are validated
-
 ###### Without having used a for loop or having filled the array manually, is the output the following?
 
 ```console
 [[ 7.  1.  7.]
-[nan  2.  2.]
-[nan  8.  8.]
-[ 9.  3.  9.]
-[ 8.  9.  8.]
-[nan  2.  2.]
-[ 8.  2.  8.]
-[nan  6.  6.]
-[ 9.  2.  9.]
-[ 8.  5.  8.]]
+ [nan  2.  2.]
+ [nan  8.  8.]
+ [ 9.  3.  9.]
+ [ 8.  9.  8.]
+ [nan  2.  2.]
+ [ 8.  2.  8.]
+ [nan  6.  6.]
+ [ 9.  2.  9.]
+ [ 8.  5.  8.]]
 ```
 
-There are two steps in this exercise:
-
-- Create the vector that contains the grade of the first exam if available or the second. This can be done using `np.where`:
+---
 
-```python
-    np.where(np.isnan(grades[:, 0]), grades[:, 1],     grades[:, 0])
-```
+---
 
-- Add this vector as third column of the array. Here are two ways:
+#### Exercise 8: Wine
 
-```python
-    np.insert(arr = grades, values = new_vector, axis = 1, obj = 2)
+###### Was the text file successfully loaded into a NumPy array using `genfromtxt('winequality-red.csv', delimiter=';')` and optimized for memory usage, weighing `76800` bytes or less?
 
-    np.hstack((grades, new_vector[:, None]))
-```
+Use this in the solution to confirm:
 
----
+```Python
 
----
+# Check the optimized data size
+optimized_size = optimized_data.nbytes
 
-#### Exercise 8: Wine
+# Verify if the dataset size criterion is met
+if optimized_size <= 76800:
+    print("Data optimized successfully.")
+else:
+    print("Optimization criteria not met.")
+```
 
-##### The exercise is validated if all questions of the exercise are validated
+##### For question 2:
 
-###### Has the text file successfully been loaded in a NumPy array with `genfromtxt('winequality-red.csv', delimiter=';')` and the reduced arrays weights **76800 bytes**?
+"Display the 2nd, 7th, and 12th rows as a two-dimensional array. Exclude `np.nan` values if present."
 
 ###### Is the output the following?
```

Comment:
The current "expected output" is not the same as the one obtained with our solution. To avoid confusion, I would remove the expected result and ask:
```
Is the output in line with the data present in the provided dataset in the subject?
```

Reply:
done